### PR TITLE
Pass alignments to Matcher callbacks

### PR DESCRIPTION
**Pull request by kwhumphreys**
## Description

When Matcher is called with `with_alignments=True`, make alignments accessible in `on_match` callback functions.

### Types of change

Enhancement to make Matcher results consistent in callbacks.

## Checklist

- [x] I have submitted the spaCy Contributor Agreement.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Review by svlandeg — COMMENTED** [review]
Hi,

Thanks for the PR, much appreciated! I get what you're trying to do and I think it makes sense to make this change. 

I was just wondering, aesthetically, whether we can avoid copy-pasting the `on_match` application by not returning the results in either of the two `if`/`elif` statements, but rather build an array `final_results` or something such that either holds the `spans` or the `matches`, and which is used to iterate over once for the `on_match` calls right at the end of the function before anything is returned? 

(I think this explanation is more lengthy than the actual chang required- you'd basically just need to rename a few parameters and remove the `return spans` line)

**Comment by kwhumphreys** [comment]
Thanks for the suggestion. I wasn't clear why alignments were being removed here https://github.com/explosion/spaCy/blob/master/spacy/matcher/matcher.pyx#L286 so I didn't want to change it originally, but it seems that the removal is not needed.

**Comment by svlandeg** [comment]
Yea I think you solved it elegantly with the `*_` bit!

File: spacy/matcher/matcher.pyx [hunk 1]
```diff
@@ -281,28 +281,19 @@ cdef class Matcher:
...
-        # perform the callbacks on the filtered set of results
-        for i, (key, start, end) in enumerate(final_matches):
-            on_match = self._callbacks.get(key, None)
-            if on_match is not None:
-                on_match(self, doc, i, final_matches)
         if as_spans:
-            spans = []
-            for key, start, end in final_matches:
+            final_results = []
+            for i, (key, start, end, *_) in enumerate(final_matches):
```
Comment (svlandeg) [inline spacy/matcher/matcher.pyx:286]:
```suggestion
            for key, start, end, *_ in final_matches:
```
It doesn't matter much, but I don't think we need to `enumerate` here?

**Pull request merged**